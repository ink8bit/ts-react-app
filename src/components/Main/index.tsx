import type { ReactElement } from 'react';

import { Routes, Route } from 'react-router-dom';
import React from 'react';

import Home from '../Home';
import About from '../About';
import s from './styles.css';

function Main(): ReactElement {
  return (
    <main className={s.main}>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="about" element={<About />} />
      </Routes>
    </main>
  );
}

export default Main;
