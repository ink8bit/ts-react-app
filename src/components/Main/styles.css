.main {
  background: lightgoldenrodyellow;
  padding: var(--offset);
}

.title {
  margin: 0;
}
