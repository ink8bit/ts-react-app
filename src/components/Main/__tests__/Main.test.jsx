import React from 'react';
import { create } from 'react-test-renderer';

import Main from '../index';

test('snapshot', () => {
  const c = create(<Main />);
  expect(c.toJSON()).toMatchSnapshot();
});
