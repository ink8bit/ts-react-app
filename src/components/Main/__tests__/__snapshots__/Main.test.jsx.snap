// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<main
  className="main"
>
  <h1
    className="title"
  >
    Main content
  </h1>
</main>
`;
